jvm-libs/extensions/src/main/kotlin/TimerExtensions.kt

```diff
@@ -0,0 +1,20 @@
+/*
+ * Copyright Consensys Software Inc.
+ *
+ * This file is dual-licensed under either the MIT license or Apache License 2.0.
+ * See the LICENSE-MIT and LICENSE-APACHE files in the repository root for details.
+ *
+ * SPDX-License-Identifier: MIT OR Apache-2.0
+ */
+package maru.extensions
+
+import net.consensys.linea.async.toSafeFuture
+import net.consensys.linea.metrics.DynamicTagTimer
+import net.consensys.linea.metrics.Timer
+import tech.pegasys.teku.infrastructure.async.SafeFuture
+
+fun <T> Timer.captureTimeSafeFuture(safeFuture: SafeFuture<T>): SafeFuture<T> =
```

Comment:
Since `CompletableFuture` is easily convertable to `SafeFuture`, why won't we convert the argument where it's used instead? Otherwise, by the same reasoning, instead of converting the `CompletableFuture`-s to `SafeFuture` we would add extensions for other usecases of `SafeFuture`, which doesn't scale well and will bloat the API.

Reply:
I would suggest to either opt for `timer.captureTime(sf.toCompletableFuture()).toSafeFuture()` or to rename the extension you added to 
```
Timer.captureTime(safeFuture: SafeFuture<T>): SafeFuture<T>
```
I don't see a reason to add the type info into the method name itself, when the type info is in the argument and the return type.

Reply:
Ah, ok. Now it makes sense